### PR TITLE
Survey styling

### PR DESCRIPTION
**Pull request by Hackman78**

### CODE REVIEW

File: client/src/components/Search.tsx [hunk 1]
```diff
@@ -8,26 +8,58 @@ import Select, { SelectChangeEvent } from '@mui/material/Select';
...
-        { id: number; name: string; tag: string }[]
-    >([]);
+    const [selectedTags, setSelectedTags] = useState<Tag[]>([]);
+    const [tags, setTags] = useState<Tag[]>([]);
+    const tagsRef = useRef(tags);
+
+    const getAllTags = () => {
+        axios.get('/api/tags')
+            .then(({ data }) => {
+                console.log(data);
```
Comment (cody-daigle) [inline client/src/components/Search.tsx]: Is this still being used?

File: client/src/components/SearchResults.tsx
Comment (cody-daigle) [inline client/src/components/SearchResults.tsx:1]: Assuming this was sent on accident?

File: client/src/components/Survey.tsx [hunk 2]
```diff
@@ -1,83 +1,101 @@
...
+	const [allPostTags, setPostTags] = useState<TagsResponse[]>([]);
+	const [selectedTags, setSelectedTags] = useState<TagsResponse[]>([]);
+	const navigate = useNavigate();
 
-  const handleSubmit = (event: React.FormEvent) => {
-    event.preventDefault();
-    // Handle form submission here
-  };
+	const handleSubmit = (selectedTags: TagsResponse[]) => {
+		console.log('hiii', selectedTags);
```
Comment (cody-daigle) [inline client/src/components/Survey.tsx]: Is this still being used?

File: client/src/components/Survey.tsx [hunk 3]
```diff
@@ -1,83 +1,101 @@
 import axios from 'axios';
 import React, { useState, useEffect } from 'react';
+import SurveyTag from './SurveyTag';
+import { redirect, useNavigate } from 'react-router-dom';
+import { Box, Button, Typography, Container, Grid, Paper } from '@mui/material';
```
Comment (cody-daigle) [inline client/src/components/Survey.tsx]: If redirect isn't being used it should be removed on **line 4**

File: client/src/components/SurveyTag.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import Chip from '@mui/material/Chip';
+import { SxProps } from '@mui/system';
+import { Theme } from '@mui/material/styles';
```
Comment (cody-daigle) [inline client/src/components/SurveyTag.tsx:4]: Lines `3` & `4` should be imported individually

File: client/src/components/Tags.tsx [hunk 5]
```diff
@@ -28,7 +28,7 @@ const Tags = () => {
 		console.log(id, 'YOU CLICKED!!');
 		try {
 			await axios.patch(`/api/tags/${id}`);
-			getUserTags();
+			await getUserTags();
```
Comment (cody-daigle) [inline client/src/components/Tags.tsx:31]: Should this be refactored to form best practices / consistency within async/await function?

**Review by cody-daigle — CHANGES_REQUESTED** [review]

File: client/src/components/Search.tsx (hunk 1 above)
Comment (Hackman78) [reply client/src/components/Search.tsx]: yes this is being used in the useEffect to grab tags for the autocomplete function

File: client/src/components/Survey.tsx (hunk 3 above)
Comment (Hackman78) [reply client/src/components/Survey.tsx]: Taken out now

File: client/src/components/Survey.tsx (hunk 2 above)
Comment (Hackman78) [reply client/src/components/Survey.tsx]: deleted now

**Review by cody-daigle — APPROVED** [review]

File: client/src/components/Drawers.tsx
Comment (AlexPHebert2000) [inline client/src/components/Drawers.tsx:1]: Did your indentation on VSCode change to 4 spaces in this file, the indents seem really large

File: client/src/components/Search.tsx [hunk 6]
```diff
@@ -8,101 +8,140 @@ import Select, { SelectChangeEvent } from '@mui/material/Select';
...
-        { id: number; name: string; tag: string }[]
-    >([]);
+	const [selectedTags, setSelectedTags] = useState<Tag[]>([]);
+	const [tags, setTags] = useState<Tag[]>([]);
+
+	const getAllTags = () => {
+		axios
+			.get('/api/tags')
+			.then(({ data }) => {
+				console.log(data);
```
Comment (AlexPHebert2000) [inline client/src/components/Search.tsx]: Remove console log

File: client/src/components/Search.tsx [hunk 7]
```diff
@@ -8,101 +8,140 @@ import Select, { SelectChangeEvent } from '@mui/material/Select';
...
-        newValue: { id: number; name: string; tag: string }[]
-    ) => {
-        setSelectedTags(newValue);
-    };
+	const handleSearch = () => {
+		const names = selectedTags.map((tag) => tag.name).join('-');
+		axios
+			.get(`/api/search/${tagType}/${names}`)
+			.then(({ data }) => {
+				console.log(data);
```
Comment (AlexPHebert2000) [inline client/src/components/Search.tsx]: Remove console log

File: client/src/components/Survey.tsx [hunk 8]
```diff
@@ -1,83 +1,105 @@
...
+	};
 
-  const getAllTags = async () => {
-    try {
-      const { data } = await axios.get('/api/tags/all');
-      // console.log('allmaps', data);
+	const getAllTags = async () => {
+		try {
+			const { data } = await axios.get('/api/tags/all');
+			console.log(data);
```
Comment (AlexPHebert2000) [inline client/src/components/Survey.tsx]: remove console log

File: client/src/components/Tags.tsx [hunk 9]
```diff
@@ -28,7 +28,7 @@ const Tags = () => {
 		// console.log(id, 'YOU CLICKED!!');
```
Comment (AlexPHebert2000) [inline client/src/components/Tags.tsx]: Remove if unnecessary

**Review by AlexPHebert2000 — CHANGES_REQUESTED** [review]
Remove logs

**Review by AlexPHebert2000 — APPROVED** [review]

**Pull request merged**